src/main/java/com/adobe/target/edge/client/service/TelemetryService.java

```diff
@@ -81,6 +100,33 @@ private TelemetryEntry createTelemetryEntry(
         .timestamp(System.currentTimeMillis());
   }
 
+  private TelemetryEntry createTelemetryEntry(
+      TargetDeliveryRequest targetDeliveryRequest,
+      TargetDeliveryResponse targetDeliveryResponse,
+      double executionTime,
+      double parsingTime,
+      long responseSize) {
+    if (!clientConfig.isTelemetryEnabled()) {
+      return null;
+    }
+
+    TelemetryFeatures telemetryFeatures = buildTelemetryFeatures(targetDeliveryRequest);
```

Comment:
I think in order to make this more DRY we could have a shared base method called by both implementations of `createTelemetryEntry` which will create TelemetryEntry, set its base properties, and return it.  Then your new `createTelemetryEntry` can just add the `parsing` and `request` properties to it.

Reply:
fixed redundant call